apps/dokuha-roadmap-frontend/src/app/details/[contentId]/page.tsx

```diff
@@ -0,0 +1,182 @@
+        return;
+      }
+
+      if (currentPage > totalPage) {
+        alert('現在のページ数は総ページ数よりも小さな値を入力してください。');
+        return;
+      }
+
+      const formatPOSTData: postData = {
+        "currentPage": Number(currentPage),
+        "note": note
+      }
+
+      const response = await fetch(`${process.env.NEXT_PUBLIC_BE_URL}learning-contents/${contentId}`, {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(formatPOSTData),
+      });
+
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      }
+
+      alert('記録が保存されました。');
+    } catch (error) {
+      // TODO: 適切なエラーハンドリングに置き換える
+      console.error('エラーが発生しました:', error);
+    }
+  };
+
+  const onBackPage = () => {
+    // TODO: 変更内容保存しないよ、みたいな警告だす。未編集の場合は出さないようにする？ 
+    router.push('/');
+  };
+
+  if (isLoading) {
+    return <div className="flex justify-center items-center h-screen">データ読み込み中...</div>;
+  }
+
+  return (
+    <div className="flex flex-col h-screen p-8 bg-gray-50 relative">
+      <div className="flex flex-col md:flex-row justify-between items-start md:items-center mb-8 gap-4">
+        <div className="flex-grow">
+          <h1 className="text-3xl font-extrabold text-gray-800">{contentTitle}</h1>
+          <p className="text-lg text-gray-600 mt-2">
+            <label htmlFor="currentPageInput">
+            現在のページ: 
+            <input
+              id="currentPageInput"
+              type="number"
+              value={currentPage}
+              onChange={(e) => {
+                const value = e.target.value;
+                setCurrentPage(value === '' ? 0 : (Number.isNaN(Number(value)) ? 0 : Number(value)));
+              }}
+              min="0"
+              max={totalPage}
+              className="ml-2 w-24 p-1 border border-gray-300 rounded-md text-center font-semibold text-blue-600 focus:outline-none focus:ring-2 focus:ring-blue-500"
+            />
+            </label>
+            <span className="ml-1"> / {totalPage}</span>
+          </p>
+        </div>
+
+        <div className="w-full md:w-1/2 lg:w-1/3 flex-shrink-0">
+          <div className="w-full bg-gray-200 rounded-full h-8 overflow-hidden">
+            <div
+              className="bg-blue-500 h-full rounded-full transition-all duration-500 ease-out flex items-center justify-center"
+              style={{ width: `${clampedProgress}%` }}
+            >
+              <span className="text-white text-sm font-bold drop-shadow-sm">
+                {clampedProgress.toFixed(0)}%
+              </span>
+            </div>
+          </div>
+          <p className="text-center text-blue-700 font-semibold mt-2">
+            {progress < 25 && '始まったばかり！コツコツ頑張ろう！'}
+            {progress >= 25 && progress < 75 && 'いい感じ！この調子で進めよう！'}
+            {progress >= 75 && progress < 100 && 'もう少しで読了！がんばれ！'}
+            {progress === 100 && 'D　O　K　U　H　A　！'}
+          </p>
```

Comment:
[IMO]
JSXの仲が結構長くなっていて、ロジックも結構記載されているから、ここは関数化すると良さそう！
この部分はこうして、[こんな感じにするイメージ](https://github.com/k-naoyuki/DOKUHA-roadmap/pull/15#discussion_r2169712397) : https://github.com/k-naoyuki/DOKUHA-roadmap/pull/15#discussion_r2169712397
```suggestion
          <p className="text-center text-blue-700 font-semibold mt-2">
            {getProgressMessage(progress)}
          </p>
```